backends/tc/ebpfCodeGen.cpp

```diff
@@ -407,14 +407,170 @@ void TCIngressPipelinePNA::emitLocalVariables(EBPF::CodeBuilder *builder) {
     builder->newline();
 }
 
+
 // =====================EBPFPnaParser=============================
+EBPFPnaParser::EBPFPnaParser(const EBPF::EBPFProgram *program, const IR::ParserBlock *block,
+                             const P4::TypeMap *typeMap)
+    : EBPFPsaParser(program, block, typeMap) {
+    visitor = new PnaStateTranslationVisitor(program->refMap, program->typeMap, this);
+}
+
 void EBPFPnaParser::emit(EBPF::CodeBuilder *builder) {
     builder->emitIndent();
     builder->blockStart();
     EBPF::EBPFParser::emit(builder);
     builder->blockEnd(true);
 }
 
+void PnaStateTranslationVisitor::compileExtractField(const IR::Expression *expr,
+                                                  const IR::StructField *field, unsigned alignment,
+                                                  EBPF::EBPFType *type) {
+    unsigned widthToExtract = dynamic_cast<EBPF::IHasWidth *>(type)->widthInBits();
+    auto program = state->parser->program;
+    cstring msgStr;
+    cstring fieldName = field->name.name;
+
+    bool checkIfMAC = false;
```

Comment:
The eBPF back end has no knowledge of "MAC" . Rather it looks like you are trying to match on a specific bit-width?

Reply:
This change is not in eBPF compiler, but in P4TC. If user gives annotation "macaddr", the expectation is not to convert the endianness. This is not on specific bit width, but annotation "macaddr".